mesonbuild/wrap/wrap.py

```diff
@@ -203,15 +203,17 @@ def get_git(self, p):
                 subprocess.call(['git', 'pull'], cwd=checkoutdir)
             else:
                 if subprocess.call(['git', 'checkout', revno], cwd=checkoutdir) != 0:
-                    subprocess.check_call(['git', 'fetch'], cwd=checkoutdir)
+                    subprocess.check_call(['git', 'fetch', p.get('url'), revno], cwd=checkoutdir)
                     subprocess.check_call(['git', 'checkout', revno],
                                           cwd=checkoutdir)
         else:
             subprocess.check_call(['git', 'clone', p.get('url'),
                                    p.get('directory')], cwd=self.subdir_root)
             if revno.lower() != 'head':
-                subprocess.check_call(['git', 'checkout', revno],
-                                      cwd=checkoutdir)
+                if subprocess.call(['git', 'checkout', revno], cwd=checkoutdir) != 0:
+                    subprocess.check_call(['git', 'fetch', p.get('url'), revno], cwd=checkoutdir)
+                    subprocess.check_call(['git', 'checkout', revno],
```

Comment:
These changes should not be in this MR as they have nothing to do with the Arm compiler as such. Please move them to their own MR.

Reply:
Yes I will back these out. I realized later that this should be a separate change. Sorry for the mess up.